ssdutil/main.py

```diff
@@ -16,10 +17,46 @@
 
 DEFAULT_DEVICE="/dev/sda"
 SYSLOG_IDENTIFIER = "ssdutil"
+DISK_TYPE_SSD = "0"
+DISK_INVALID = "-1"
 
 # Global logger instance
 log = logger.Logger(SYSLOG_IDENTIFIER)
 
+def get_default_disk():
+    """Check default disk"""
+    default_device = DEFAULT_DEVICE
+    host_mnt = '/host'
+    cmd = "lsblk -l -n |grep {}".format(host_mnt)
+    proc = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE)
+    out = proc.stdout.readline()
+    if host_mnt in out:
+        dev_nums = out.split()[1]
+        dev_maj_num = dev_nums.split(':')[0]
+
+        cmd = "lsblk -l -I {} |grep disk".format(dev_maj_num)
+        proc = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE)
+        out = proc.stdout.readline()
+        if "disk" in out:
+            default_device = os.path.join("/dev/", out.split()[0])
+
+    return default_device
+
+
+def get_disk_type(diskdev):
+    """Check disk type"""
+    diskdev_name = diskdev.replace('/dev/','')
+    cmd = "lsblk -l -n |grep {}".format(diskdev_name)
+    proc = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE)
+    out = proc.stdout.readline()
+    if diskdev_name not in out:
+        return DISK_INVAILD
+    cmd = "cat /sys/block/{}/queue/rotational".format(diskdev_name)
+    proc = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE)
```

Comment:
Forking to `cat` here seems unnecessary. Is there a reason you chose not to use native python io here?

Reply:
It is also more pythonic to do this and add error handling in order to handle the `DISK_INVALID` case rather than forking `lsblk` as well.